src/BuildIntegration/Microsoft.NETCore.Native.Windows.props

```diff
@@ -80,12 +80,18 @@ See the LICENSE file in the project root for more information.
       <Output TaskParameter="ExitCode" PropertyName="_WhereCppTools"/>
     </Exec>
 
-    <Exec Condition="'$(_WhereCppTools)' != '0'" Command="&quot;$(MSBuildThisFileDirectory)findvcvarsall.bat&quot; x86_amd64" LogStandardErrorAsError="true" ContinueOnError="false" ConsoleToMSBuild="true">
-      <Output TaskParameter="ConsoleOutput" PropertyName="_CppToolsDirectory"/>
+    <Exec Condition="'$(_WhereCppTools)' != '0'" Command="&quot;$(MSBuildThisFileDirectory)findvcvarsall.bat&quot; amd64" LogStandardErrorAsError="true" ContinueOnError="false" ConsoleToMSBuild="true" StandardOutputImportance="Low">
+      <Output TaskParameter="ConsoleOutput" PropertyName="_FindVCVarsallOutput"/>
       <Output TaskParameter="ExitCode" PropertyName="_VCVarsAllFound"/>
     </Exec>
 
+    <ItemGroup Condition="'$(_VCVarsAllFound)' == '0'">
+      <AdditionalNativeLibraryDirectories Include="$(_FindVCVarsallOutput.Split(`#`)[1].Split(';'))" />
+      <CustomLinkerArg Include="/LIBPATH:&quot;%(AdditionalNativeLibraryDirectories.Identity)&quot;" />
```

Comment:
This would be better to do in the targets file so that the app's .csproj file can add to AdditionalNativeLibraryDirectories as well if needed.

Reply:
By moving `CustomLinkerArg` to targets file, test projects are failing.

In `Microsoft.NETCore.Native.targets`, there is:

```xml
<Import
  Project="$(MSBuildThisFileDirectory)Microsoft.NETCore.Native.Windows.props"
  Condition="'$(TargetOS)' == 'Windows_NT'" />
```

Seems like this condition is not met when running tests and therefore `Microsoft.NETCore.Native.Windows.props` is not imported. However, host OS condition is satisfied for test projects:

```xml
<CustomLinkerArg
  Include="/LIBPATH:&quot;%(AdditionalNativeLibraryDirectories.Identity)&quot;"
  Condition="'$(OS)' == 'Windows_NT'" />
```

If I manually apply the PR changes to `%USERPROFILE%\.nuget\packages\microsoft.dotnet.ilcompiler\1.0.0-alpha-26208-01\build`, it compiles the WebAPI.

Should the test runner be fixed so it sets `HostOS` to `WindowsNT`?

Reply:
We will either need to break the convention `CoreRT_<env-var-name>` if we need to include it: https://github.com/dotnet/corert/blob/099a02aad7b358a72f9e92d01a05dc1ae26c8adc/tests/runtest.cmd#L8-L12 or use `CoreRT_HostOS` and set `<HostOS>` in a new `dir.props` (there isn't any `dir.props` in `tests/src/Simple/` atm).

Reply:
Never mind, fixed by adding `'@(AdditionalNativeLibraryDirectories->Count())' &gt; 0` condition. 😄